metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Taxonomy.java

```diff
@@ -76,16 +105,86 @@ public long getTaxonomyId() {
         return taxonomyId;
     }
 
+    public void setTaxonomyId(long taxonomyId) {
+        this.taxonomyId = taxonomyId;
+    }
+
     public String getName() {
         return name;
     }
 
+    public void setName(String name) {
+        this.name = name;
+    }
+
     /**
      * Release date control: taxonomies are always public.
      */
     @Override
+    @JsonIgnore
     public LocalDate getReleaseDate() {
         return null;
     }
 
+    public Rank getRank() {
+        return rank;
+    }
+
+    public void setRank(Rank rank) {
+        this.rank = rank;
+    }
+
+    @ApiModelProperty(position = 5, readOnly = true)
+    public Long getTaxonomySpeciesId() {
+        return (taxonomySpecies == null) ? -1 : taxonomySpecies.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 7, readOnly = true)
+    public Long getTaxonomyGenusId() {
+        return (taxonomyGenus == null) ? -1 : taxonomyGenus.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 9, readOnly = true)
+    public Long getTaxOrderId() {
+        return (taxonomyOrder == null) ? -1 : taxonomyOrder.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 11, readOnly = true)
+    public Long getTaxonomyClassId() {
+        return (taxonomyClass == null) ? -1 : taxonomyClass.getTaxonomyId();
+    }
+
+    public Taxonomy getTaxonomySpecies() {
+        return taxonomySpecies;
+    }
+
+    public void setTaxonomySpecies(Taxonomy taxonomySpecies) {
+        this.taxonomySpecies = taxonomySpecies;
+    }
+
+    public Taxonomy getTaxonomyGenus() {
+        return taxonomyGenus;
+    }
+
+    public void setTaxonomyGenus(Taxonomy taxonomyGenus) {
+        this.taxonomyGenus = taxonomyGenus;
+    }
+
+    public Taxonomy getTaxonomyOrder() {
+        return taxonomyOrder;
+    }
+
+    public void setTaxonomyOrder(Taxonomy taxonomyOrder) {
+        this.taxonomyOrder = taxonomyOrder;
+    }
+
+    public Taxonomy getTaxonomyClass() {
+        return taxonomyClass;
+    }
+
+    public void setTaxonomyClass(Taxonomy taxonomyClass) {
+        this.taxonomyClass = taxonomyClass;
+    }
+
+    public enum Rank {SUBSPECIES, SPECIES, GENUS, ORDER, CLASS, UNKNOWN}
```

Comment:
If I remember correctly we decided to not have an enum and to just store rank in plain text as represented in NCBI taxonomy tree

Reply:
yes, we thought that, but anyways this enum can be a standard for searching the taxonomy by rank, the empty ranks will put in to UNKNOWN.

Reply:
We don't want the empty rank to be "UNKNOWN", it should represent exactly what the NCBI taxonomy says it to be (be it "species", "subspecies", "strain" or "no rank"). In theory we could create a huge ENUM with all possible NCBI taxonomy ranks, but it would be a pain to constantly maintain and synchronise it, so frankly I don't see the point. Just storing rank as a string is totally fine.

Reply:
For the purposes of search by class/order/etc. it also won't be a problem because references to fixed ranks are stored in the appropriate columns already; i. e. we don't need to know that a certain taxonomy is of RANK.CLASS rank to be able to query by class, because it is already referenced to by the "class" column

Reply:
okies then I can remove it